packages/types/src/Metadata/v2/latest.substrate.v2.json

```diff
@@ -0,0 +1,5687 @@
+            },
+            {
+              "name": "AllExtrinsicsLen",
+              "modifier": "Optional",
+              "type": {
+                "PlainType": {
+                  "typeName": "u32",
+                  "displayName": "u32"
+                }
+              },
+              "fallback": "0x00",
+              "docs": []
+            },
+            {
+              "name": "BlockHash",
+              "modifier": "Default",
+              "type": {
+                "MapType": {
+                  "key": {
+                    "typeName": "u64",
+                    "displayName": "BlockNumber"
+                  },
+                  "value": {
+                    "typeName": "H256",
+                    "displayName": "Hash"
+                  }
+                }
+              },
+              "fallback": "0x0000000000000000000000000000000000000000000000000000000000000000",
+              "docs": []
+            },
+            {
+              "name": "ExtrinsicData",
+              "modifier": "Default",
+              "type": {
+                "MapType": {
+                  "key": {
+                    "typeName": "u32",
+                    "displayName": "u32"
+                  },
+                  "value": {
+                    "typeName": "Vec<u8>",
+                    "displayName": "Bytes"
+                  }
+                }
+              },
+              "fallback": "0x00",
+              "docs": []
+            },
+            {
+              "name": "RandomSeed",
+              "modifier": "Default",
+              "type": {
+                "PlainType": {
+                  "typeName": "H256",
+                  "displayName": "Hash"
+                }
+              },
+              "fallback": "0x0000000000000000000000000000000000000000000000000000000000000000",
+              "docs": []
+            },
+            {
+              "name": "Number",
+              "modifier": "Default",
+              "type": {
+                "PlainType": {
+                  "typeName": "u64",
+                  "displayName": "BlockNumber"
+                }
+              },
+              "fallback": "0x0000000000000000",
+              "docs": [
+                " The current block number being processed. Set by `execute_block`."
+              ]
+            },
+            {
+              "name": "ParentHash",
+              "modifier": "Default",
+              "type": {
+                "PlainType": {
+                  "typeName": "H256",
+                  "displayName": "Hash"
+                }
+              },
+              "fallback": "0x0000000000000000000000000000000000000000000000000000000000000000",
+              "docs": []
+            },
+            {
+              "name": "ExtrinsicsRoot",
+              "modifier": "Default",
+              "type": {
+                "PlainType": {
+                  "typeName": "H256",
+                  "displayName": "Hash"
```

Comment:
I did the minimal amount of work to get it work (which is already a lot) so there are definitely rooms for optimizations.
The question is, is this a dealbreaker, or something can be improved later?

Reply:
i would suggest leave optimizations until we can get rid of legacy v0/v1. I did a lot of compromises in this PR because some codes in type-storage and type-extrinsic are bonded to v0's implementation. Otherwise I feel we're not able to improve more.